cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -132,6 +132,16 @@ public ClassLoader getArtifactClassLoader(ArtifactId artifactId) throws IOExcept
   }
 
   /**
+   * Create a new instance of plugin class without any config, config will be null in the instantiated plugin.
+   * @param pluginType Type of plugin class
+   * @param <T> Type of plugin
+   * @return a new plugin instance
+   */
+  public <T> T newInstanceWithoutConfig(TypeToken<?> pluginType) {
```

Comment:
kind of a weird public method to have, it means I have to have the type already? Seems like it should be passing in Plugin, or the ArtifactId and PluginClass, or something like that.